### ISSUE TITLE
Issue with Writable Streams

### ISSUE THREAD

**Issue by zpappa** [neumino/rethinkdbdash#255]
**When using a writable stream to insert data, I am 'skipping' some records**
**If I don't use a stream and just insert data via 'insert', then all records are processed**
There doesn't seem to be a pattern to it, it's different records that are skipped every time.

I wrote the below to serve as an example.

Am I doing something wrong or is this an issue?

**Non stream example**
```
function importCollection(collection) {
    var file = JSON.parse(fs.readFileSync('remapped_output/'+collection+'.json'));
    r.tableCreate(collection).then(function () {
        for(var i=0;i<file.length;i++) {
            var doc = file[i];
            r.table(collection).insert(doc).then(function(result) {
                console.log(result);
            });
        }
        console.log("wrote "+i+" documents to collection "+collection);
    });
}
```

vs 

**Stream example**
```
function importCollection(collection) {
    var file = JSON.parse(fs.readFileSync('remapped_output/'+collection+'.json'));
    r.tableCreate(collection).then(function () {
        var table = r.table(collection).toStream({writable: true});
        for(var i=0;i<file.length;i++) {
            var doc = file[i];
            table.write(doc, function(result) {
                console.log(JSON.stringify(result));
            });
        }
        console.log("wrote "+i+" documents to collection "+collection);
    });
}
```

**Comment by neumino** [comment]
Why do you think you're skipping data?

Did you run a read after?

**Comment by zpappa** [comment]
Yes, I did it with small collections, less than 300 documents and larger ones with 30k plus documents. In both cases my count differed. On the 300 document collection I lost about 100 documents.

If I do insert via the non stream method my counts are perfect every time.

**Comment by zpappa** [comment]
I also originally had it streaming from a file but it was harder to see where the error was, so I am doing the JSON parse from a sync file read and I debugged it and verified the count.

**Comment by neumino** [comment]
Hum, I can reproduce that, but I'm not sure what the problem is. I'll try to look at it this week end.

**Comment by zpappa** [comment]
Thanks, I will take a look as well. I'm in love with being able to do this with streams and I have an impl using highland on top that is working in other areas and I'd like to continue it. 

PS thanks for responding so quickly.

**Comment by neumino** [comment]
Fixed in 2.3.7
